src/axom/core/Macros.hpp

```diff
@@ -79,10 +79,12 @@
  */
 #ifdef AXOM_USE_CUDA
   #define AXOM_LAMBDA [=] AXOM_HOST_DEVICE
+  #define AXOM_MUTABLE_LAMBDA [=] AXOM_HOST_DEVICE
   #define AXOM_DEVICE_LAMBDA [=] AXOM_DEVICE
   #define AXOM_HOST_LAMBDA [=] AXOM_HOST
 #else
   #define AXOM_LAMBDA [=]
+  #define AXOM_MUTABLE_LAMBDA [&]
   #define AXOM_DEVICE_LAMBDA [=]
   #define AXOM_HOST_LAMBDA [=]
 #endif
```

Comment:
I added a new macro: `AXOM_MUTABLE_LAMBDA` as discussed elsewhere in this PR.

We should discuss the necessity and/or name for this new macro.

Reply:
Note: nvcc does not allow the following: ` [&] AXOM_HOST_DEVICE `

Reply:
**Update:** I removed the `AXOM_MUTABLE_LAMBDA` macro from this file in favor of a local solution in `MeshTester_detail.hpp`